python/whylogs/extras/image_metric.py

```diff
@@ -211,6 +213,9 @@ def _update_relevant_submetrics(self, name: str, data: PreprocessedColumn) -> No
     def columnar_update(self, view: PreprocessedColumn) -> OperationResult:
         count = 0
         for image in list(chain.from_iterable(view.raw_iterator())):
+            if isinstance(image, np.ndarray):
+                image = Image.fromarray(image.astype(np.uint8))
```

Comment:
do you need the cast there? uint8 is common, but there is also 'F' and 'I'* modes, see https://pillow.readthedocs.io/en/latest/handbook/concepts.html#modes. If we restrict the support to uint8 modes that might be ok, but we would need to document that limitation and help warn when people send in other kinds of inputs.

For example you can send pixels as 16 bit integer values in "I;16" in which case the expected range of values could go into the thousands.

Reply:
It won't work without the cast.

```python
import numpy as np

img2 = np.array(img).tolist()
Image.fromarray(np.array(img2))
```

That will  throw with `TypeError: Cannot handle this data type: (1, 1, 3), <i8`. That only happens when you're starting from a nested python list.

Reply:
Got it, lets scope this to uint8 image data initially, then what you have should be reasonable.